server/client.go

```diff
@@ -1556,6 +1616,15 @@ func (c *client) deliverMsg(sub *subscription, mh, msg []byte) bool {
 		return false
 	}
 
+	// Check if we have a subscribe deny clause. This will trigger us to check the subject
+	// for a match against the denied subjects.
+	if client.mperms != nil {
+		if client.checkDenySub(string(c.pa.subject)) {
```

Comment:
Could we make it a one-line statement?

Reply:
yes good call, added it for performance testing.